### DISCUSSION TITLE
Backreference population when eager loading

### DISCUSSION THREAD

**Discussion by Metadorius** [linq2db/linq2db#3773] [Q&A] [answered]
Suppose you have table `A` and table `B` which references `A` in one-to-many relationship. If you eager load stuff from `A` with stuff from `B` via (pseudocode) `.LoadWith(a => a.bs)`, you can access your `B`s normally using `a.bs`, and then if you try to access `b.a` - the reference will be null. Is there any way to set that reference on eager load automatically?

Though maybe it's some kind of a problem with my set up, because my hierarchy is a bit more complex than that with three levels and for the second level referenced object on the first level is not being set.

**Comment by viceroypenguin** [comment, marked as answer]
Currently, there is no way to do both directions on the same query. You can either query the parents and `.LoadWith()` the children, or query the children and `.LoadWith()` the parents. If you need bidirectional links, you'll have to take the downloaded data and add the references separately.

**Reply by jods4** [reply]
Should be pointed out that fixing all references in an arbitrarily deep graph would require an entity tracker (i.e. a map of id -> entity for everything loaded), which linq2db doesn't have as it is not an ORM.

Maybe interceptors allow this to be done in user-land? I'm not sure if it's possible but it would be cool.
